packages/rpc-core/src/rpc-methods/getSupply.ts

```diff
@@ -0,0 +1,49 @@
+import { Base58EncodedAddress } from '@solana/keys';
+import { Commitment, LamportsUnsafeBeyond2Pow53Minus1, RpcResponse } from './common';
+
+type GetSupplyCommonConfig = Readonly<{
+    commitment?: Commitment;
+}>;
+
+type GetSupplyApiResponseBase = RpcResponse<{
+    /** Total supply in lamports */
+    total: LamportsUnsafeBeyond2Pow53Minus1;
+    /** Circulating supply in lamports */
+    circulating: LamportsUnsafeBeyond2Pow53Minus1;
+    /** Non-circulating supply in lamports */
+    nonCirculating: LamportsUnsafeBeyond2Pow53Minus1;
+}>;
+
+type GetSupplyApiResponseWithNonCirculatingAccounts = GetSupplyApiResponseBase &
+    Readonly<{
+        value: Readonly<{
+            /** an array of account addresses of non-circulating accounts */
+            nonCirculatingAccounts: [Base58EncodedAddress];
+        }>;
+    }>;
+
+type GetSupplyApiResponseWithoutNonCirculatingAccounts = GetSupplyApiResponseBase &
+    Readonly<{
+        value: Readonly<{
+            nonCirculatingAccounts: [];
+        }>;
+    }>;
+
+export interface GetSupplyApi {
+    /**
+     * Returns information about the current supply.
+     */
+    getSupply(
+        config: GetSupplyCommonConfig &
+            Readonly<{
+                excludeNonCirculatingAccountsList: true;
+            }>
+    ): GetSupplyApiResponseWithoutNonCirculatingAccounts;
+    getSupply(
+        config: GetSupplyCommonConfig &
+            Readonly<{
+                excludeNonCirculatingAccountsList: false;
+            }>
+    ): GetSupplyApiResponseWithNonCirculatingAccounts;
+    getSupply(config?: GetSupplyCommonConfig): GetSupplyApiResponseWithNonCirculatingAccounts;
```

Comment:
Pretty sure you only have to supply an overload for the case where `excludeNonCirculatingAccountsList` is `true`.

```suggestion
type GetSupplyConfig = Readonly<{
    commitment?: Commitment;
    excludeNonCirculatingAccountsList?: boolean;
}>;

type GetSupplyApiResponseBase = RpcResponse<{
    /** Total supply in lamports */
    total: LamportsUnsafeBeyond2Pow53Minus1;
    /** Circulating supply in lamports */
    circulating: LamportsUnsafeBeyond2Pow53Minus1;
    /** Non-circulating supply in lamports */
    nonCirculating: LamportsUnsafeBeyond2Pow53Minus1;
}>;

type GetSupplyApiResponseWithNonCirculatingAccounts = GetSupplyApiResponseBase &
    Readonly<{
        value: Readonly<{
            /** an array of account addresses of non-circulating accounts */
            nonCirculatingAccounts: [Base58EncodedAddress];
        }>;
    }>;

type GetSupplyApiResponseWithoutNonCirculatingAccounts = GetSupplyApiResponseBase &
    Readonly<{
        value: Readonly<{
            nonCirculatingAccounts: [];
        }>;
    }>;

export interface GetSupplyApi {
    /**
     * Returns information about the current supply.
     */
    getSupply(
        config: GetSupplyConfig &
            Readonly<{
                excludeNonCirculatingAccountsList: true;
            }>
    ): GetSupplyApiResponseWithoutNonCirculatingAccounts;
    getSupply(config?: GetSupplyConfig): GetSupplyApiResponseWithNonCirculatingAccounts;
```

Reply:
Thanks, that's way cleaner!